### DISCUSSION TITLE
about "is being rate limited"

### DISCUSSION THREAD

**Discussion by mlonV** [vectordotdev/vector#16579] [Q&A] [answered]
About "is being rate limited"
source: "file"
sink: loki
Starting a vector into loki always causes this error. Why?

Feb 24 20:23:18 test-server vector[25183]:  2023-02-24T12:23:18.555798Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=25}: vector_common::internal_event::service:  Internal log [Service call failed. No retries or retries exhausted.] is being rate limited.
Feb 24 20:23:18 test-server vector[25183]:  2023-02-24T12:23:18.555822Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=25}: vector_common::internal_event::component_events_dropped:  Internal log [Events dropped] is being rate limited.
Feb 24 20:23:28 test-server vector[25183]:  2023-02-24T12:23:28.070274Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=32}: vector::sinks::util::retries:  Internal log [Non-retriable error; dropping the request.] has been rate limited 7 times.
Feb 24 20:23:28 test-server vector[25183]:  2023-02-24T12:23:28.070314Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=32}: vector::sinks::util::retries: Non-retriable error;  dropping the request. error=Server responded with an error: 400 Bad Request internal_log_rate_limit=true
Feb 24 20:23:28 test-server vector[25183]:  2023-02-24T12:23:28.070350Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=32}: vector_common::internal_event::service:  Internal log [Service call failed. No retries or retries exhausted.] has been rate limited 7 times.
Feb 24 20:23:28 test-server vector[25183]:  2023-02-24T12:23:28.070358Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=32}: vector_common::internal_event::service:  Service call failed. No retries or retries exhausted. error=Some(ServerError { code:  400 }) request_id=32 error_type="request_failed" stage="sending" internal_log_rate_limit=true
Feb 24 20:23:28 test-server vector[25183]:  2023-02-24T12:23:28.070391Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=32}: vector_common::internal_event::component_events_dropped:  Internal log [Events dropped] has been rate limited 7 times.
Feb 24 20:23:28 test-server vector[25183]:  2023-02-24T12:23:28.070398Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=32}: vector_common::internal_event::component_events_dropped:  Events dropped intentional=false count=9 reason="Service call failed. No retries or retries exhausted."  internal_log_rate_limit=true
Feb 24 20:23:29 test-server vector[25183]:  2023-02-24T12:23:29.086355Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=33}: vector::sinks::util::retries:  Internal log [Non-retriable error; dropping the request.] is being rate limited.
Feb 24 20:23:29 test-server vector[25183]:  2023-02-24T12:23:29.086422Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=33}: vector_common::internal_event::service:  Internal log [Service call failed. No retries or retries exhausted.] is being rate limited.
Feb 24 20:23:29 test-server vector[25183]:  2023-02-24T12:23:29.086447Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=33}: vector_common::internal_event::component_events_dropped:  Internal log [Events dropped] is being rate limited.

**Comment by jszwedko** [comment, marked as answer]
Hi @mlonV ,

`is being rate limited` is just talking about Vector's internal logs that are being written to stderr. Vector is not rate limiting the stream of events flowing through it.

In your case, the `loki` sink is failing to send at least some events to Loki though. You can see the HTTP 400 responses there. I would check your Loki logs to see if indicates why the requests are failing.

**Reply by mlonV** [reply]
Thank you elder brother, should be loki configuration problem, trouble elder brother
limits_config:
    unordered_writes

**Reply by jszwedko** [reply]
Ah, yes, we recommend having Loki accept out-of-order writes. It should be the default behavior on newer versions of Loki.

**Reply by mlonV** [reply]
Although out-of-order writes have been set, the restriction is still received
loki also has no error
 

2023-02-27T10:48:14.799450Z  INFO source{component_kind="source" component_id=victory_phpapp component_type=file component_name=victory_phpapp}:file_server: vector::internal_events::file::source: Found new file to watch. file=/data/logs/app_log-app-2023-02-27-00.log
2023-02-27T10:48:19.869808Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=14}: vector::sinks::util::retries: Non-retriable error; dropping the request. error=Server responded with an error: 400 Bad Request internal_log_rate_limit=true
2023-02-27T10:48:19.869922Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=14}: vector_common::internal_event::service: Service call failed. No retries or retries exhausted. error=Some(ServerError { code: 400 }) request_id=14 error_type="request_failed" stage="sending" internal_log_rate_limit=true
2023-02-27T10:48:19.869979Z ERROR sink{component_kind="sink" component_id=loki component_type=loki component_name=loki}:request{request_id=14}: vector_common::internal_event::component_events_dropped: Events dropped intentional=false count=666 reason="Service call failed. No retries or retries exhausted." internal_log_rate_limit=true

**Reply by mlonV** [reply]
Hi bro
See if this has anything to do with it  （internal_log_rate_secs=10）
[root@test-server vector]# vector -c vector.toml
2023-02-28t05.59:19.261921z INFO vector::app: Internal log rate limit configured.  internal_log_rate_secs=10

**Reply by jszwedko** [reply]
It surprises me that Loki wouldn't log any errors when it returns a 400. I'd double check if there are additional logging parameters on the Loki side that could show more detail.

**Reply by mlonV** [reply]
Thank you very much. I think I also need to find out whether it is loki's problem

**Comment by mlonV** [comment]
Maybe it's my loki's limitation or rpc error, but I don't know what's wrong


level=warn ts=2023-02-27T02:37:57.942100344Z caller=grpc_logging.go:43 method=/logproto.Pusher/Push duration=1.148304ms err="rpc error: code = Code(400) desc = entry with timestamp 2023-02-26 16:30:04 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:04 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:04 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:04 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:05 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:05 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:05 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:05 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:05 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\nentry with timestamp 2023-02-26 16:30:05 +0000 UTC ignored, reason: 'entry out of order' for stream: {hostname=\"test-server\", level=\"DEBUG\", service=\"victory.phpapp_dev\"},\ntotal ignored: 789 out of 789" msg=gRPC
